iwyu_test_util.py

```diff
@@ -503,6 +508,42 @@ def _CheckPrerequisites(cc_file):
     raise unittest.SkipTest(', '.join(failed))
 
 
+def _ParseExpectedFailures(cc_file):
+  """ Parses expected failure conditions out of cc_file. """
+  conditions = []
+  with open(cc_file) as it:
+    for lineno, line in enumerate(it):
+      m = _IWYU_TEST_XFAIL_RE.match(line)
+      if not m:
+        continue
+
+      feature = m.group(1)
+      value = m.group(2)
+
+      feature_cond = _FEATURES.known.get(feature)
+      if not feature_cond:
+        raise SyntaxError('%s:%s IWYU_XFAIL: unsupported feature: %s' %
+                          (cc_file, lineno, feature))
+
+      condition = lambda arg=value: feature_cond(arg)
+
+      desc = '%s(%s)' % (feature, value)
+      conditions.append((condition, desc))
+    return conditions
+
+
+def IsTestExpectedToFail(cc_file):
+  """ Checks if a test is expected to fail """
```

Comment:
Google style is really strict about punctuation, so let's use a full stop after `fail`